### PR TITLE
Add text barrage

### PR DESCRIPTION
**Pull request by singlecoder**

### CODE REVIEW

File: playground/text-barrage.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  "World",
+  "text",
+  "Text",
+  "ABCabc",
+  "abcDEF",
+  "abcdefg",
+  "hijklmn",
+  "opqrst",
+  "uvwxyz",
+  "ABCDEFG",
```
Comment (GuoLei1990) [inline playground/text-barrage.ts:35]: a lot of no meaning worlds, add  `peace` and `love` etc.

File: playground/text-barrage.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+  stop() {
+    this._isPlayging = false;
+  }
+}
+
+class TextManager extends Script {
+  public halfWidth: number = 0;
+  public halfHeight: number = 0;
+  public cd: number = 5;
```
Comment (GuoLei1990) [inline playground/text-barrage.ts:109]: cd is what? try use more clear name

File: playground/text-barrage.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  onUpdate(dt: number): void {
+    if (this._isPlayging) {
+      const { position } = this.entity.transform;
+      const curPosX = position.x;
+      const targetPosX = curPosX + this.speed * dt;
+      position.x = targetPosX;
+      if (targetPosX < this.range) {
+        this.reset();
+      }
+    } else {
```
Comment (GuoLei1990) [inline playground/text-barrage.ts:68]:
```ts
const { position } = this.entity.transform;
      position.x += this.speed * deltaTime;
      if (position.x < this.range) {
        this.reset();
      }
```

File: playground/text-barrage.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  return min + Math.round(rand * range);
+}
+
+class TextBarrageAnimation extends Script {
+  public halfWidth: number = 0;
+  public halfHeight: number = 0;
+  public textCount: number = 0;
+  public speed: number = 0;
+  public range: number = 0;
+  public delayFrame: number = 0;
```
Comment (GuoLei1990) [inline playground/text-barrage.ts:49]: delay what?

**Review by GuoLei1990 — APPROVED** [review]
+1

**Review by zhuxudong — APPROVED** [review]
+1

File: playground/text-barrage.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    this._textRenderer = this.entity.getComponent(TextRenderer);
+    this._range = -camera.orthographicSize * camera.aspectRatio;
+    this._reset(true);
+  }
+
+  onUpdate(dt: number): void {
+    if (this._isPlayging) {
+      const { position } = this.entity.transform;
+      position.x += this._speed * dt;
+      if (position.x < this._range) {
```
Comment (zhuxudong) [inline playground/text-barrage.ts:36]: can it use 'renderer.isCulled'

**Review by zhuxudong — APPROVED** [review]

File: playground/text-barrage.ts (hunk 5 above)
Comment (singlecoder) [reply playground/text-barrage.ts:36]: > Won't fix

**Pull request merged**